src/app/SERVICE/tasks.service.ts

```diff
@@ -1,37 +1,30 @@
 import { Injectable } from '@angular/core';
 import { Task } from '../models/task.model';
-import { Observable, map, switchMap } from 'rxjs';
+import { Observable } from 'rxjs';
 import { HttpClient } from '@angular/common/http';
-import { Router } from '@angular/router';
 import { tasksApiUrl } from '../utils/constants';
 
+
 @Injectable({
   providedIn: 'root'
 })
 export class TasksService {
 
-  constructor(private http: HttpClient, private router: Router) {
-  }
+  constructor(private http: HttpClient) { }
 
   getAllTasks(): Observable<Task[]> {
     return this.http.get<Task[]>(tasksApiUrl)
-      .pipe(
-        map(data => data)
-      );
   }
 
   createNewTask(task: Task) {
     if (task) {
-      this.http.post(tasksApiUrl, task).subscribe()
-    } else {
-      return
-    }
+      return this.http.post<Task[]>(tasksApiUrl, task)
+    } else return
```

Comment:
there shouldn't be an if statement here, since it should always create a new task as the data is passed from the completed form